vignettes/netCDF_Functionality.Rmd

```diff
@@ -0,0 +1,400 @@
+[CF conventions](http://cfconventions.org/Data/cf-conventions/cf-conventions-1.8/cf-conventions.html#_orthogonal_multidimensional_array_representation).
+
+Additionally, this document describes the Dryland Ecology Lab's standards and
+conventions in regards to naming, metadata, and netCDF structure.
+
+# Input Data Format and Structure
+
+## Variable Data
+
+The netCDF functions in this package expect an array, in which each row represents
+data for a site. Columns represent one of three things:
+
+* multiple different variables for one site
+* one variable with multiple measurements (time series)
+* one variable with multiple depth measurements
+
+```{r, create1d, results = 'hide'}
+   # load rSW2analysis pacckage
+    library(rSW2analysis)
+
+   # dummy data for 10 sites, 10 data points (could be 10 vars, 10 times, or 10 depths)
+   someData <- rnorm(100, 7, 30)
+   data1d <- array(someData, c(10, 10))
+   str(data1d)
+   is.array(data1d)
+```
+
+Additionally, it is possible to have data at multiple times *and* multiple depths.
+In this case, time should be organized by column, and multiple depths should be organized
+sequentially in third dimension, forming a 3d array.
+
+```{r, create3d, results = 'hide'}
+  # dummy data for 10 sites, 10 times, 3 depths.
+  data3d <- array(0, c(10, 10, 3))
+  data3d[,,1] <- someData
+  data3d[,,2] <- someData
+  data3d[,,3] <- someData
+```
+
+## Location Data
+
+Locations all need to be on the same coordinates reference system (CRS). Data can
+be either be organized on a regularly spaced grid (i.e. wall to wall simulations for
+a given area) or not (i.e. project data representing plot points in a park).
+
+If the former, the coordinates need to be in order. If the later,
+
+```{r, createlocs, results = 'hide'}
+  # points on a regular spaced grid in the WGS84 CRS
+  locationsGrid <- data.frame(X_WGS84 = c(rep(-124.5938, 5), rep(-124.5312, 5)),
+                        Y_WGS84 = rep(c(47.90625, 47.96875, 48.03125, 48.09375,
+                                     48.15625), 2))
+
+  # irregularly spaced points in the WGS84 CRS
+  #locationsRandom
+```
+
+## Dimension Data
+
+NetCDFs are organized into dimensions. Dimensions have two key components: Bounds
+and values. The dimensions also have other information which is degined in the Attibute Data
+(see below).
+
+### Latitude (Y) and Longitude (X) Dimensions
+
+All NetCDFs will have at least two dimensions, latitude and longitude.
+
+The bounds and values of the latitude and longitude are determined from the location
+data within the *create_empty_netCDF_file* function.
+
+### Time (T) Dimension
+
+The user must define the bounds and values for the time dimension, as inputs to the
+*create_empty_netCDF_file*.
+
+The bounds of the time dimension are the start and end of each interval of the time dimension.
+
+The values refer to the midpoint of each interval of the time dimension.
+
+The bounds and values need to be in the same units.
```

Comment:
I still think we need to highlight that our function expects a vector instead of a matrix for the bounds, despite the fact that bounds are two-dimensional; and we should explain that the expected organization of that vector is `c(start1, end1, start2, end2, ..., startn, endn)` despite the repetition due to `endk == startk+1` -- and not `c(start1, end1, end2, ..., endn)`

Reply:
fixed

Reply:
I don't understand what `The bounds of the time dimension are the start and end of each interval of the time dimension.` means without knowing what it should mean.

What about: "The boundary values for a continuous time coordinate are given by the concatenated vector of start/end pairs of each time interval, i.e., c(start1, end1, start2, end2, ..., startn, endn)."

Reply:
fixed.